### PR TITLE
Feat/i18n for docs wechaty

### PR DESCRIPTION
**Pull request by hcfw007**
rework doc wechaty section
add i18n support
https://github.com/wechaty/docusaurus/issues/1512

### CODE REVIEW

**Review by huan — CHANGES_REQUESTED** [review]
LGTM

This PR looks promising!

Please:

1. update the PR to the latest main branch and make sure it can pass the CI
2. paste some screenshots in the PR description (Issue will be better) so that we can see what it looks like after it has been merged!

**Comment by hcfw007** [comment]
This PR is ready to be reviewed. It will fullly focus on i18n part. I'll open another PR for doc update.

**Review by windmemory — DISMISSED** [review]
LGTM

**Review by su-chang — APPROVED** [review]

**Review by su-chang — DISMISSED** [review]

File: docusaurus/docusaurus.config.js [hunk 1]
```diff
@@ -28,7 +28,18 @@ const config = {
   themeConfig,
   presets,
   plugins,
-  themes,
```
Comment (huan) [inline docusaurus/docusaurus.config.js:31]: Why do we remote the `themes` variable from the list?

Comment (hcfw007) [reply docusaurus/docusaurus.config.js:31]: That seems to be a typo and it's now reverted.

**Review by huan — DISMISSED** [review]
LGTM

Thank you very much for adding the i18n solution to our docs!

**Comment by huan** [comment]
copilot:all

**Comment by huan** [comment]
This PR will be good to go after the CI has turned green. Thanks!

**Comment by hcfw007** [comment]
This CI error never occur in my local env. In local there is lots of network issue which make it hard to debug. I'll see what I can do.

[ERROR] Error: request to https://cdn.jsdelivr.net/npm/wechaty-grpc@next/dist/esm/out/wechaty/puppet.swagger.json failed, reason: read ECONNRESET
    at BaseResolver.<anonymous> (/Users/myonlystarcn/LocalDocuments/code/work/other/wechaty-homepage-doc/docusaurus/node_modules/redoc/node_modules/@redocly/openapi-core/lib/resolve.js:119:23)
    at Generator.throw (<anonymous>)
    at rejected (/Users/myonlystarcn/LocalDocuments/code/work/other/wechaty-homepage-doc/docusaurus/node_modules/redoc/node_modules/@redocly/openapi-core/lib/resolve.js:6:65)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Comment by huan** [comment]
As my investigation, the CI error is the following:

```
Warning:  Duplicate routes found!
- Attempting to create page at /zh/search, but a page already exists at this route.
This could lead to non-deterministic routing behavior.
[info] [webpackbar] Compiling Client
[info] [webpackbar] Compiling Server
[success] [webpackbar] Client: Compiled successfully in 1.[63](https://github.com/wechaty/docusaurus/actions/runs/5069018602/jobs/9102050392#step:7:64)m
[success] [webpackbar] Server: Compiled successfully in 2.00m
[info] [webpackbar] Compiling Service Worker
Error:  Unable to build website for locale zh.
Error:  Error: The route must start with the baseUrl "/zh/", but was "/qrcode/*". This is a bug, please report it.
```

Which is related to the duplicated routes, and the URL conflict problems.

**Comment by hcfw007** [comment]
@huan I have fixed CI. This is due to the qrcode directly add route. When i18n cames in, it checks the base url. 
According to this issue https://github.com/facebook/docusaurus/issues/5844

> However, when creating alternate URLs, the theme assumes that every page would have a localized base URL (which makes sense!)

**Review by windmemory — APPROVED** [review]
LGTM

**Review by huan — APPROVED** [review]
LGTM

Thank you very much for adding i18n support to our docs website, and make the CI green!

**Comment by hcfw007** [comment]
Great! Although we have only one page with Chinese version now, i'm looking forward to translate more.

**Pull request merged**